File: python/mod_8/task_2.py
def main():
    health = 1.0
    while health > 0:
        resist = float(input("Введите показатель резиста (от 0 до 1): "))
        while resist > 1 or resist < 0:
            resist =float(input("Введите число больше 0 и меньше 1: "))

        fireball = float(input("Введите урон огненного шара (от 0 до 1): "))
        while fireball > 1 or fireball < 0:
            fireball =float(input("Введите число больше 0 и меньше 1: "))

        damage = fireball * resist
        health = health - damage

        if health > 0:
            print(f"Нанесено {damage} урона! У орка осталось {health:.2f} здоровья")
        else: print(f"Нанесено {damage} урона! Вы победили орка!")

if __name__ == "__main__":
    main()

File: final/mod_8/task_2.cpp
#include <iostream>

int main(){
    using namespace std;
    float health = 1, resist = 0, fireball = 0, damage;
    cout << "=====Immolate Improved!=====";

    while (health > 0) {
        do {
            cout << "Введите показатель резиста (от 0 до 1): "; cin >> resist;
            if (resist >= 1 || resist <= 0) cout << "Введите число больше 0 и меньше 1!\n";
        } while (resist >= 1 || resist <= 0);
        
        do {
            cout << "Введите урон огненного шара (от 0 до 1): "; cin >> fireball;
            if (fireball >= 1 || fireball <= 0) cout << "Введите число больше 0 и меньше 1!\n";
        } while (fireball >= 1 || fireball <= 0);

        damage = fireball * resist;
        health -= damage;

        if (health <= 0) {
            cout << "Нанесено " << damage << " урона! Вы победили орка!\n";
        } else cout << "Нанесено " << damage << " урона! У орка осталось " << health << " здоровья.\n";
    }
}